spec/mongoid/timestamps_spec.rb

```diff
@@ -107,4 +108,270 @@
+    end
+
+    after do
+      Timecop.return
+    end
+
+    context "when touch: true" do
+      let(:user) { TimestampsSpec::Touch::User.create! }
+      let(:address) { TimestampsSpec::Touch::Address.create! }
+      let(:account) { TimestampsSpec::Touch::Account.create! }
+      let(:pet) { TimestampsSpec::Touch::Pet.create! }
+
+      before do
+        [user, address, account, pet]
+        update_time
+      end
+
+      context "when HABTM association" do
+
+        context "when updating the association itself" do
+          before do
+            user.update(addresses: [address])
+          end
+
+          it "updates the timestamps correctly" do
+            user.updated_at.should == update_time
+            address.updated_at.should == update_time
+          end
+        end
+
+        context "when updating the association's foreign key" do
+          before do
+            user.update(address_ids: [address.id])
+          end
+
+          it "updates the timestamps correctly" do
+            user.updated_at.should == update_time
+            address.updated_at.should == start_time
+          end
+        end
+      end
+
+      context "when has_many association" do
+
+        context "when updating the association itself" do
+          before do
+            user.update(accounts: [account])
+          end
+
+          it "updates the timestamps correctly" do
+            user.updated_at.should == update_time
+            account.updated_at.should == update_time
+          end
+        end
+
+        context "when updating the association's foreign key" do
+          before do
+            user.update(account_ids: [account.id])
+          end
+
+          it "updates the timestamps correctly" do
+            user.updated_at.should == update_time
+            account.updated_at.should == start_time
+          end
+        end
+      end
+
+      context "when belongs_to association; on has_many" do
+
+        context "when updating the association itself" do
+          before do
+            account.update(user: user)
+          end
+
+          it "updates the timestamps correctly" do
+            user.updated_at.should == update_time
+            account.updated_at.should == update_time
+          end
+        end
+
+        context "when updating the association's foreign key" do
+          before do
+            account.update(user_id: user.id)
+          end
+
+          it "updates the timestamps correctly" do
+            user.updated_at.should == start_time
+            account.updated_at.should == update_time
+          end
+        end
+      end
+
+      context "when has_one association" do
+
+        context "when updating the association itself" do
+          before do
+            user.update(pet: pet)
+          end
+
+          it "updates the timestamps correctly" do
+            user.updated_at.should == update_time
+            pet.updated_at.should == update_time
```

Comment:
I think all of these tests need to `reload` before inspecting `updated_at`.

Reply:
Can you explain why?

My understanding of the reload is it only needs to be used if outside sources modify the data. If mongoid modifies the data, a reload shouldn't be necessary. Is this a correct understanding?

Additionally, rails does not require a reload when updating the `updated-at`